### PR TITLE
added rewrite function

### PR DESCRIPTION
**Pull request by nradz**
copy_blob function doesn't works for large files. Now, you can select to use "copy_blob" original function or to use "rewrite" function. "rewrite" function can work with large files.

### CODE REVIEW

File: SwissKnife/gcloud/GCloudStorage.py [hunk 1]
```diff
@@ -277,20 +278,64 @@ def copy_blob(self, src_blob: Blob, dst_storage: "GCloudStorage", dst_file_name:
...
-        destination_bucket = dst_storage.bucket 
-        dst_path = dst_storage.get_storage_complete_file_path(file_name=dst_file_name, with_prefix=True, with_bucket=False, with_gs=False)
-
-        dst_blob = self.bucket.copy_blob(src_blob, destination_bucket, dst_path)
+        if mode == "copy":
+            dst_blob = self.__copy_blob_using_copy_function(src_blob, dst_storage, dst_file_name)
+        elif mode == "rewrite":
+            dst_blob = self.__copy_blob_using_rewrite_function(src_blob, dst_storage, dst_file_name)
+        else:
+            raise RuntimeError(f"Invalid mode for copy_blob method: ${mode}")
```
Comment (ju4nlu) [inline SwissKnife/gcloud/GCloudStorage.py]: Is the `$` written on purpose?

File: SwissKnife/gcloud/GCloudStorage.py [hunk 2]
```diff
@@ -277,20 +278,64 @@ def copy_blob(self, src_blob: Blob, dst_storage: "GCloudStorage", dst_file_name:
...
+        :type src_blob: Blob
+        :param dst_storage: The destination GCloudStorage object.
+        :type dst_storage: GCloudStorage
+        :param dst_file_name: The destination file name.
+        :type dst_file_name: str
+        :return: The destination blob. 
+        :rtype: Blob
+        """
+
+        dst_blob = dst_storage.__generate_blob("", dst_file_name, encoding=None)
```
Comment (ju4nlu) [inline SwissKnife/gcloud/GCloudStorage.py:335]: What's the meaning of the empty string that is passed as first argument?

**Review by ju4nlu — APPROVED** [review]
LGTM, just a couple of questions with no real importance.

As a side note, I think we DO really need to set up a code formatter, such as [black](https://github.com/psf/black), some functions are really hard to follow due to the number of parameters and lack of formatting that we currently have.

File: SwissKnife/gcloud/GCloudStorage.py [hunk 3]
```diff
@@ -277,20 +278,64 @@ def copy_blob(self, src_blob: Blob, dst_storage: "GCloudStorage", dst_file_name:
...
         complete_src_path = self.get_storage_complete_file_path(file_name=src_blob.name, with_bucket=True, with_prefix=True, with_gs=True)
         complete_dst_path = dst_storage.get_storage_complete_file_path(file_name=dst_blob.name, with_bucket=True, with_prefix=True, with_gs=True) 
+
+        # Log complete paths
         self.logger.info(f"Blob {complete_src_path} copied to blob {complete_dst_path}")
 
         return dst_blob
+    
+    def __copy_blob_using_copy_function(self, src_blob: Blob, dst_storage: "GCloudStorage", dst_file_name: str) -> Blob:
+        """Copies a blob using the copy_blob function. For large files, using the "rewrite" version is better.
```
Comment (eoterogarcia) [inline SwissKnife/gcloud/GCloudStorage.py:302]: `For large files, "rewrite" version is better.`, it sounds better (I think)

File: SwissKnife/gcloud/GCloudStorage.py [hunk 4]
```diff
@@ -267,7 +268,7 @@ def list_blobs(self,
                                                           with_gs=False)
         return self.bucket.list_blobs(prefix=list_prefix)
 
-    def copy_blob(self, src_blob: Blob, dst_storage: "GCloudStorage", dst_file_name: str) -> Blob:
```
Comment (eoterogarcia) [inline SwissKnife/gcloud/GCloudStorage.py:270]: To be more specific with `dst_storage` parameter, you can import `from google.cloud.storage.bucket import Bucket` and then specify type in `dst_storage` like this: `dst_storage: Bucket`

**Review by eoterogarcia — APPROVED** [review]

File: SwissKnife/gcloud/GCloudStorage.py (hunk 2 above)
Comment (eoterogarcia) [reply SwissKnife/gcloud/GCloudStorage.py:335]: It is the gcloud storage path of the file, but it is already in `dst_storage` object, am I right?

**Comment by eoterogarcia** [comment]
> LGTM, just a couple of questions with no real importance.
> 
> As a side note, I think we DO really need to set up a code formatter, such as [black](https://github.com/psf/black), some functions are really hard to follow due to the number of parameters and lack of formatting that we currently have.

I totally agree with @ju4nlu on the code formatter. We can even include it with github actions.

File: SwissKnife/gcloud/GCloudStorage.py (hunk 1 above)
Comment (nradz) [reply SwissKnife/gcloud/GCloudStorage.py]: Nope, it's a mistake.

File: SwissKnife/gcloud/GCloudStorage.py (hunk 2 above)
Comment (nradz) [reply SwissKnife/gcloud/GCloudStorage.py:335]: For example, if you want to write in "some_folder/some_file.csv", the first param will be "some_folder" and the second param "some_file.csv". In this case, the function receives a single complete path. To avoid split it, I use it directly on the second parameter.

File: SwissKnife/gcloud/GCloudStorage.py (hunk 4 above)
Comment (nradz) [reply SwissKnife/gcloud/GCloudStorage.py:270]: That param is not a bucket from Google Library. It is the GCloudStorage from SwissKnife.
I can't use it as type hinting from itself (this is a method of GCloudStorage object). For that reason, I used a explicit string.

Comment (eoterogarcia) [reply SwissKnife/gcloud/GCloudStorage.py:270]: True! I was confused with the gcloud copy function... sorry for the misunderstanding!

File: SwissKnife/gcloud/GCloudStorage.py (hunk 3 above)
Comment (nradz) [reply SwissKnife/gcloud/GCloudStorage.py:302]: :+1:
